java/src/hu/holyoil/crewmate/Robot.java

```diff
@@ -76,6 +74,14 @@ public void ReactToAsteroidExplosion() {
         onAsteroid.GetRandomNeighbour().ReactToMove(onAsteroid, this);
     }
 
+    @Override
+    public void Drill() {
+        super.Drill();
+        if (onAsteroid.GetLayerCount() == 0) {
+            Logger.Log(this, "I've revealed a(n) " + onAsteroid.GetResource().toString() + " on " + onAsteroid.GetId());
```

Comment:
This can produce a nullpointer if the asteroid has no resource

Reply:
hopp tényleg